test/filter.test.ts

```diff
@@ -83,13 +83,31 @@ describe("FilterCheck", () => {
 			Optional(Number.withConstraint((n) => n > 5))
 		);
 
-		expect(checkWrapper(10)).toEqual(10);
+		expect(checkWrapper(10)).toBe(10);
+		expect(checkWrapper(undefined)).toBeUndefined();
 
 		const checkChain = FilterCheck(
 			Number.optional().withConstraint((n) => typeof n === "undefined" || n > 5)
 		);
 
-		expect(checkChain(10)).toEqual(10);
+		expect(checkChain(10)).toBe(10);
+		expect(checkChain(undefined)).toBeUndefined();
+
+		const checkRecord = FilterCheck(
+			Record({
+				a: Number.withConstraint((n) => n > 5).optional(),
+			}).optional()
+		);
+
+		expect(checkRecord({ a: 10 })).toStrictEqual({ a: 10 });
+		expect(checkRecord({})).toStrictEqual({});
+		expect(checkRecord(undefined)).toBeUndefined();
+
+		const checkUnion = FilterCheck(Union(Number, Null).optional());
+
+		expect(checkUnion(10)).toBe(10);
+		expect(checkUnion(null)).toBeNull();
+		expect(checkUnion(undefined)).toBeUndefined();
```

Comment:
On `main` this would throw:

```typescript
TypeError: Cannot read property 'reflect' of undefined

  34 |      }
  35 | 
> 36 |      const r = t.reflect;
     |                  ^
  37 | 
  38 |      if (isNotImplemented(r)) {
  39 |              throw new Error(`Type "${r.tag}" is not filterable`);

  at filterInternal (src/index.ts:36:14)
  at filterInternal (src/index.ts:82:11)
  at filterInternal (src/index.ts:85:11)
  at filter (src/index.ts:96:9)
  at src/index.ts:158:16
  at Object.<anonymous> (test/filter.test.ts:110:10)
```